packages/internal/src/generate/plugins/rw-typescript-resolvers/index.ts

```diff
@@ -0,0 +1,142 @@
+
+  const optionalSignForInfoArg = visitor.config.optionalInfoArgument ? '?' : ''
+
+  defsToInclude.push(
+    'export type Resolver<TResult, TParent = {}, TContext = {}, TArgs = {}> = ' +
+      'ResolverFn<TResult, TParent, TContext, TArgs>'
+  )
+
+  const header = `${indexSignature}
+
+${visitor.getResolverTypeWrapperSignature()}
+
+${defsToInclude.join('\n')}
+
+export type SubscriptionSubscribeFn<TResult, TParent, TContext, TArgs> = (
+  parent: TParent,
+  args: TArgs,
+  context: TContext,
+  info${optionalSignForInfoArg}: GraphQLResolveInfo
+) => AsyncIterable<TResult> | Promise<AsyncIterable<TResult>>;
+
+export type SubscriptionResolveFn<TResult, TParent, TContext, TArgs> = (
+  parent: TParent,
+  args: TArgs,
+  context: TContext,
+  info${optionalSignForInfoArg}: GraphQLResolveInfo
+) => TResult | Promise<TResult>;
+
+export interface SubscriptionSubscriberObject<TResult, TKey extends string, TParent, TContext, TArgs> {
+  subscribe: SubscriptionSubscribeFn<{ [key in TKey]: TResult }, TParent, TContext, TArgs>;
+  resolve?: SubscriptionResolveFn<TResult, { [key in TKey]: TResult }, TContext, TArgs>;
+}
+
+export interface SubscriptionResolverObject<TResult, TParent, TContext, TArgs> {
+  subscribe: SubscriptionSubscribeFn<any, TParent, TContext, TArgs>;
+  resolve: SubscriptionResolveFn<TResult, any, TContext, TArgs>;
+}
+
+export type SubscriptionObject<TResult, TKey extends string, TParent, TContext, TArgs> =
+  | SubscriptionSubscriberObject<TResult, TKey, TParent, TContext, TArgs>
+  | SubscriptionResolverObject<TResult, TParent, TContext, TArgs>;
+
+export type SubscriptionResolver<TResult, TKey extends string, TParent = {}, TContext = {}, TArgs = {}> =
+  | ((...args: any[]) => SubscriptionObject<TResult, TKey, TParent, TContext, TArgs>)
+  | SubscriptionObject<TResult, TKey, TParent, TContext, TArgs>;
+
+export type TypeResolveFn<TTypes, TParent = {}, TContext = {}> = (
+  parent: TParent,
+  context: TContext,
+  info${optionalSignForInfoArg}: GraphQLResolveInfo
+) => ${namespacedImportPrefix}Maybe<TTypes> | Promise<${namespacedImportPrefix}Maybe<TTypes>>;
+
+export type IsTypeOfResolverFn<T = {}, TContext = {}> = (obj: T, context: TContext, info${optionalSignForInfoArg}: GraphQLResolveInfo) => boolean | Promise<boolean>;
+
+export type NextResolverFn<T> = () => Promise<T>;
+
+export type DirectiveResolverFn<TResult = {}, TParent = {}, TContext = {}, TArgs = {}> = (
+  next: NextResolverFn<TResult>,
+  parent: TParent,
+  args: TArgs,
+  context: TContext,
+  info${optionalSignForInfoArg}: GraphQLResolveInfo
+) => TResult | Promise<TResult>;
+`
+
+  const resolversTypeMapping = visitor.buildResolversTypes()
+  const resolversParentTypeMapping = visitor.buildResolversParentTypes()
+  const { getRootResolver, getAllDirectiveResolvers, hasScalars } = visitor
+
+  if (hasScalars()) {
+    imports.push('GraphQLScalarType', 'GraphQLScalarTypeConfig')
+  }
+
+  return {
+    ...basePlugin(schema, [], config),
```

Comment:
I'm spreading `prepend` and `content` from the original plugin here. `content` will be fully overwritten by our own `content` below. Really only need our own `visitorResult`, but couldn't figure out how to be more granular than this.